### DISCUSSION TITLE
HTTP 메서드 중 GET 메서드와 POST 메서드에 대해 설명해주세요.

### DISCUSSION THREAD

**Discussion by yeonkkk** [woowacourse-study/2023-tal-study#63] [기술 면접 질답 모음집]
HTTP 메서드 중 GET 메서드와 POST 메서드에 대해 설명하고, 차이점에 대해 설명해주세요.

**Comment by yeonkkk** [comment]
`GET` 메서드는 클라이언트에서 서버로 특정 리소스 정보를 요청하기 위해 사용되는 메서드이고, `POST` 메서드는는 리소스를 생성하거나 업데이트하기 위해 사용하는 메서드 입니다.    

두 메서드의 차이는 크게 3 가지를 생각해볼 수 있습니다.

먼저 GET 요청에 대한 응답은 캐시가 가능하고, POST 요청에 대한 응답은 캐시 되지 않습니다.  

그리고 GET 요청 시 전달할 데이터를 쿼리 스트링을 사용하며 데이터 길이 제한이 있습니다. 
반면에 POST 요청 시 리퀘스트 바디로 데이터를 전달하고 데이터 길이 제한이 없습니다.  

마지막으로 GET 요청은 멱등성을 가지고, POST는 멱등성을 가지지 않습니다.

**Comment by yoondgu** [comment]
GET 메서드는 서버로부터 자원을 조회할 때, POST 메서드는 자원을 새로 생성하거나 업데이트 때 사용한다.

GET 메서드는 주로 조회 요청을 하며 함께 보낼 데이터를 쿼리 스트링으로 보내고, POST 메서드는 민감한 정보를 url에 노출시키지 않도록 body로 보내는 경우가 많다.

하지만 2014년도 RFC 문서에서 GET 메서드의 body는 무시해야 한다는 내용이 삭제되면서, 스펙의 해석에 따라 GET 메서드에서도 Body에 값을 보내는 경우도 많아졌다. 쿼리 스트링으로 보낼 조회 기준이 무수히 많은 경우를 고려해서다. (단, 일부 클라이언트에서는 지원되지 않을 수 있다.)

그 외 차이점으로는, GET 요청은 캐시되고 POST 요청은 캐시된다는 것 / POST는 멱등하지 못하다는 것 등이 있다.
